src/middlewared/middlewared/plugins/init_shutdown_script.py

```diff
@@ -132,27 +145,50 @@ class Config:
     async def execute_task(self, task):
         task_type = task['type']
         cmd = None
+        tmp_script = None
 
         if task_type == 'COMMAND':
             cmd = task['command']
+        elif task_type == 'SCRIPT' and task['script_text']:
+            _, tmp_script = tempfile.mkstemp(text=True)
+            os.chmod(tmp_script, stat.S_IRWXU)
+            with open(tmp_script, 'w') as f:
+                f.write(task['script_text'])
+            cmd = f'exec {tmp_script}'
         elif os.path.exists(task['script'] or '') and os.access(task['script'], os.X_OK):
             cmd = f'exec {task["script"]}'
 
-        if cmd:
-            proc = await Popen(
-                cmd,
-                stdout=subprocess.DEVNULL,
-                stderr=subprocess.PIPE,
-                shell=True,
-                close_fds=True
-            )
-            stdout, stderr = await proc.communicate()
-
-            if proc.returncode:
-                self.middleware.logger.debug(
-                    'Execution failed for '
-                    f'{task_type} {task["command"] if task_type == "COMMAND" else task["script"]}: {stderr.decode()}'
+        try:
+            if cmd:
+                proc = await Popen(
+                    cmd,
+                    stdout=subprocess.DEVNULL,
+                    stderr=subprocess.PIPE,
+                    shell=True,
+                    close_fds=True
                 )
+                stdout, stderr = await proc.communicate()
+
+                if proc.returncode:
+                    if task_type == 'COMMAND':
+                        cmd = task['command']
+                    elif task_type == 'SCRIPT' and task['script_text']:
+                        cmd = task['script_text']
+                    elif task_type == 'SCRIPT' and task['script']:
+                        cmd = task['script']
+                    else:
+                        cmd = ''
+                    self.middleware.logger.debug(
+                        'Execution failed for '
+                        f'{task_type} {cmd}: {stderr.decode()}'
+                    )
+        except Exception as error:
+            self.middleware.logger.debug(
+                f'{task["type"]} {cmd}: {error.strerror}'
```

Comment:
`Exception` class does not have `strerror` attribute, it would be better to use `{error!r}`